### ISSUE TITLE
transformation-matrix.d.ts incorrect usage of rest operator syntax

### ISSUE THREAD

**Issue by 0v3rst33r** [chrvadala/transformation-matrix#55]
`TypeScript` does not like this:
```
export function fromDefinition(...definitionOrArrayOfDefinition: MatrixDescriptor | MatrixDescriptor[]): Matrix[];
```
It borks with
```
TS2370: A rest parameter must be of an array type
```
Perhaps use method overloading?
```
  export function fromDefinition(definition: MatrixDescriptor): Matrix[];
  export function fromDefinition(...arrayOfDefinition: MatrixDescriptor[]): Matrix[];
```
or just remove the `...` rest parameter syntax?

I had to roll back to `2.0.4` in order for this pkg to compile in my build pipe line.

**Comment by mcwebb** [comment]
Sorry this was my fault - I've raised a new PR (#57) to fix this.

Would also appreciate feedback on my note in the PR description regarding the fact that any descriptors passed will need to be explicitly typed.

**Comment by chrvadala** [comment]
Fixes with 2.0.6
